### ISSUE TITLE
Crash on server logon

### ISSUE THREAD

**Issue by raymondbh** [Nuchaz/BiblioCraft#287]
The Client crash when I try to connect to my fresh server. Could be other addons fault, but yours is mentioned in the log.

Edit:
I forgot to mention that this is for the BiblioWoods TFC edition.
In TFC 0.79.18 you should not load anvil recipes, since the server superseeds those... I have made a pull request on another addons aswell...
It goes something like this:
raymondbh/TFC-Cellars-Addon@8c5da44
https://github.com/raymondbh/TFC-Cellars-Addon/commit/e293047ba358676e8bba9507736cfb8fb3420d8a

[spoiler]
[14:35:39] [Client thread/INFO]: Connecting to tfc.interburn.org., 25205
[14:35:43] [Client thread/FATAL]: Reported exception thrown!
net.minecraft.util.ReportedException: Ticking screen
	at net.minecraft.client.Minecraft.func_71407_l(Minecraft.java:1675) ~[bao.class:?]
	at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:973) ~[bao.class:?]
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:898) [bao.class:?]
	at net.minecraft.client.main.Main.main(SourceFile:148) [Main.class:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.7.0_80]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.7.0_80]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.7.0_80]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.7.0_80]
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135) [launchwrapper-1.11.jar:?]
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28) [launchwrapper-1.11.jar:?]
Caused by: java.lang.NullPointerException
	at com.bioxx.tfc.Core.TFC_Core.getSuperSeed(TFC_Core.java:1393) ~[TFC_Core.class:?]
	at com.bioxx.tfc.api.Crafting.AnvilRecipe.<init>(AnvilRecipe.java:30) ~[AnvilRecipe.class:?]
	at jds.bibliowood.tfcwood.WoodsLoader.initAnvilRecipies(WoodsLoader.java:345) ~[WoodsLoader.class:?]
	at jds.bibliowood.tfcwood.AnvilRecipeHandler.onLoadWorld(AnvilRecipeHandler.java:13) ~[AnvilRecipeHandler.class:?]
	at cpw.mods.fml.common.eventhandler.ASMEventHandler_119_AnvilRecipeHandler_onLoadWorld_Load.invoke(.dynamic) ~[?:?]
	at cpw.mods.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:54) ~[ASMEventHandler.class:?]
	at cpw.mods.fml.common.eventhandler.EventBus.post(EventBus.java:138) ~[EventBus.class:?]
	at net.minecraft.client.multiplayer.WorldClient.<init>(WorldClient.java:61) ~[bjf.class:?]
	at net.minecraft.client.network.NetHandlerPlayClient.func_147282_a(NetHandlerPlayClient.java:238) ~[bjb.class:?]
	at net.minecraft.network.play.server.S01PacketJoinGame.func_148833_a(SourceFile:70) ~[hd.class:?]
	at net.minecraft.network.play.server.S01PacketJoinGame.func_148833_a(SourceFile:13) ~[hd.class:?]
	at net.minecraft.network.NetworkManager.func_74428_b(NetworkManager.java:212) ~[ej.class:?]
	at net.minecraft.client.multiplayer.GuiConnecting.func_73876_c(SourceFile:78) ~[bcx.class:?]
	at net.minecraft.client.Minecraft.func_71407_l(Minecraft.java:1661) ~[bao.class:?]

[/spoiler]

**Comment by raymondbh** [comment]
I maybe forgot to mention that this is for the BiblioWoods TFC edition.
In TFC 0.79.18 you should not load anvil recipes, since the server superseeds those... I have med a pull request on another addons aswell...
It goes something like this:
https://github.com/raymondbh/TFC-Cellars-Addon/commit/8c5da444d683046b3a2b98fde24583aa3076557c

**Comment by Nuchaz** [comment]
I will try to get a fix out for this very soon. I was just investigating this yesterday, but wasn't sure what was going on. Thanks for sharing that link to the cellars addon. I actually used that as my original reference for getting anvil recipes to work, so I couldn't have asked for a better example on what needs to change. ;)

**Comment by raymondbh** [comment]
If you need some help testing, I can assist. :-)

**Comment by raymondbh** [comment]
It was a lot more to it than I thought :P Checkout the new pull request I made for the Cellars addon:
https://github.com/Sladki/TFC-Cellars-Addon/pull/3

(And and all credits to Aleksey-Terzi, the code is mostly his - https://github.com/Aleksey-Terzi)

**Comment by Nuchaz** [comment]
Thanks for the update on that. I wouldn't have been aware that more changes were needed otherwise. I'll go over the code and try to make it happen on my end.

**Comment by Nuchaz** [comment]
So this implementation seems to work, but I am having an issue with the ServerTickHandler. It seems to be running every tick which is causing a good bit of lag. Esp when there is metal on the anvil, my test server is hitting like 120ms+ / tick (up from ~5ms). It doesn't seem that it is actually needed though. I commented out my line that registers the ServerTickHandler and the lag goes away and the anvil recipes still seem to work. It seems the ChunkEventHandler setup is adding the recipe to the server no problems and just need the initWorldPacket / PlayerTracker setup for the client side recipes. I'll do some more testing to make sure.

**Comment by raymondbh** [comment]
Did you add a test to break the adding of recipies if it's already added?

Check line 188-193 https://github.com/raymondbh/TFC-Cellars-Addon/blob/master/src/java/sladki/tfc/ModManager.java

**Comment by Nuchaz** [comment]
yeah, I have that in released versions as well. There is 12 test cases it has to run though. Either way, it isn't something that should run every tick. Ideally it should only run on world load and on player joining server. 

I just did some testing outside my dev environment with a compiled version and it seems to work fine without the ServerTickHandler setup. I tested both on an SMP server (Only newest TFC, BiblioCraft, and BiblioWoods TFC edition installed) and singleplayer with no issues so far. I'm out of time tonight, but I plan to test a couple more things tomorrow and if all goes well I'll try to push out a release.
